lib/buffer.js

```diff
@@ -379,6 +378,16 @@ Buffer.prototype.toString = function(encoding, start, end) {
 };
 
 
+Buffer.prototype.toString = function() {
+  const length = this.length | 0;
+  if (length === 0)
+    return '';
+  if (arguments.length === 0)
+    return this.utf8Slice(0, length);
+  return slowToString.apply(this, arguments);
```

Comment:
Ok, true.

In my local microbenchmark `function(encoding, start, end) {` and `return slowToString.call(this, encoding, start, end)` wins for all number of arguments.